### PR TITLE
Add MultiOperation frontend validation

### PR DESCRIPTION
**Pull request by stephanos**
## What changed?

Added Frontend validation for MultiOperationExecution requests.

What _hasn't_ changed (yet): History validation and per-operation Frontend validation.

## Why?

To make sure MultiOperationExecution request is valid before passing it to History.

## How did you test it?

Added tests.

### CODE REVIEW

File: service/frontend/errors.go [hunk 1]
```diff
@@ -137,3 +147,62 @@ var (
...
+	// this should never happen, but it's better to set it to `Aborted` than to panic
+	if code == nil {
+		c := codes.Aborted
+		code = &c
+	}
+
+	st := status.New(*code, e.Error())
+	st, _ = st.WithDetails(statuses...)
+	return st
+}
```
Comment (stephanos) [inline service/frontend/errors.go:208]: See https://github.com/temporalio/api-go/pull/158

File: service/frontend/workflow_handler.go
Comment (stephanos) [inline service/frontend/workflow_handler.go:1]: NOTE: the validation of the Start and Update operation themselves requires a bit more refactoring and will be provided in a follow-up PR.

Comment (stephanos) [inline service/frontend/workflow_handler.go:1]: NOTE: the validation of the Start and Update operation themselves requires a bit more refactoring and will be provided in a follow-up PR.

File: service/frontend/errors.go [hunk 2]
```diff
@@ -137,3 +149,62 @@ var (
 
 	errListHistoryTasksNotAllowed = serviceerror.NewPermissionDenied("ListHistoryTasks feature is disabled on this cluster.", "")
 )
+
+// TODO: copied from `api-go` to make this PR compile
```
Comment (stephanos) [inline service/frontend/errors.go:153]: See https://github.com/temporalio/api-go/pull/158

File: service/frontend/workflow_handler.go
Comment (stephanos) [inline service/frontend/workflow_handler.go:1]: NOTE: the validation of the Start and Update operation themselves requires a bit more refactoring and will be provided in a follow-up PR.

Comment (stephanos) [inline service/frontend/workflow_handler.go:1]: NOTE: the validation of the Start and Update operation themselves requires a bit more refactoring and will be provided in a follow-up PR.

File: .github/workflows/golangci-lint.yml [hunk 3]
```diff
@@ -23,6 +23,7 @@ jobs:
           # When updating the version, update the update-linters target in Makefile as well.
           version: v1.57.2
           args: --verbose --timeout 10m --fix=false --new-from-rev=HEAD~ --config=.golangci.yml
+          skip-pkg-cache: true
```
Comment (stephanos) [inline .github/workflows/golangci-lint.yml:26]: As discussed internally, the lint step was causing problems regularly - this seems to fix it.

**Review by alexshtin — APPROVED** [review]

**Pull request merged**